src/lib.rs

```diff
@@ -97,21 +105,23 @@ trait_set! {
    pub trait NeptuneField = PrimeField + ec_gpu::GpuName;
 }
 
+#[cfg(feature = "std")]
 mod serde_impl;
 
 pub(crate) const TEST_SEED: [u8; 16] = [
     0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc, 0xe5,
 ];
 
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
```

Comment:
Did you try serde's `no_std` support? 
https://serde.rs/no-std.html#no-std-support

Reply:
I did not and we don't need it for now, so I just feature gated it. But nice catch, didn't know serde has this option. Nonetheless, not gonna use it.